src/test/rpc/DepositAuthorized_test.cpp

```diff
@@ -276,11 +285,246 @@ class DepositAuthorized_test : public beast::unit_test::suite
+        }
+
+        {
+            testcase(
+                "deposit_authorized with credentials failed: not a credential "
+                "index");
+
+            auto args = depositAuthArgs(
+                alice,
+                becky,
+                "validated",
+                {"0127AB8B4B29CCDBB61AA51C0799A8A6BB80B86A9899807C11ED576AF8516"
+                 "473"});
+
+            auto const jv =
+                env.rpc("json", "deposit_authorized", args.toStyledString());
+
+            auto const& result{jv[jss::result]};
+            BEAST_EXPECT(result[jss::status] == jss::error);
+            BEAST_EXPECT(result[jss::error_code] == rpcBAD_CREDENTIALS);
+        }
+
+        {
+            testcase(
+                "deposit_authorized with credentials not authorized: "
+                "credential not accepted");
+            auto const jv = env.rpc(
+                "json",
+                "deposit_authorized",
+                depositAuthArgs(alice, becky, "validated", {credIdx})
+                    .toStyledString());
+            auto const& result{jv[jss::result]};
+            BEAST_EXPECT(result[jss::status] == jss::error);
+            BEAST_EXPECT(result[jss::error_code] == rpcBAD_CREDENTIALS);
+        }
+
+        // alice accept credentials
+        env(credentials::accept(alice, carol, credType));
+        env.close();
+
+        {
+            testcase("deposit_authorized with duplicates in credentials");
+            auto const jv = env.rpc(
+                "json",
+                "deposit_authorized",
+                depositAuthArgs(alice, becky, "validated", {credIdx, credIdx})
+                    .toStyledString());
+            auto const& result{jv[jss::result]};
+            BEAST_EXPECT(result[jss::status] == jss::error);
+            BEAST_EXPECT(result[jss::error_code] == rpcBAD_CREDENTIALS);
+        }
+
+        {
+            static const std::vector<std::string> credIds = {
+                "18004829F915654A81B11C4AB8218D96FED67F209B58328A72314FB6EA288B"
+                "E4",
+                "28004829F915654A81B11C4AB8218D96FED67F209B58328A72314FB6EA288B"
+                "E4",
+                "38004829F915654A81B11C4AB8218D96FED67F209B58328A72314FB6EA288B"
+                "E4",
+                "48004829F915654A81B11C4AB8218D96FED67F209B58328A72314FB6EA288B"
+                "E4",
+                "58004829F915654A81B11C4AB8218D96FED67F209B58328A72314FB6EA288B"
+                "E4",
+                "68004829F915654A81B11C4AB8218D96FED67F209B58328A72314FB6EA288B"
+                "E4",
+                "78004829F915654A81B11C4AB8218D96FED67F209B58328A72314FB6EA288B"
+                "E4",
+                "88004829F915654A81B11C4AB8218D96FED67F209B58328A72314FB6EA288B"
+                "E4",
+                "98004829F915654A81B11C4AB8218D96FED67F209B58328A72314FB6EA288B"
+                "E4"};
+            assert(credIds.size() > maxCredentialsArraySize);
```

Comment:
Can this be a `static_assert`, since the variable is a `static const`?

Reply:
No, vector doesn't support constexpress methods